substratools/workspace.py

```diff
@@ -190,7 +133,13 @@ def __init__(
                 makedir_safe(d)
 
 
-class AggregateAlgoWorkspace(AlgoWorkspace):
-    """Filesystem workspace for aggregate algo execution."""
+class AlgoWorkspace(GenericAlgoWorkspace):
+    """Filesystem workspace for algo execution."""
 
-    pass
+
+class CompositeAlgoWorkspace(GenericAlgoWorkspace):
+    """Filesystem workspace for composite algo execution."""
+
+
+class AggregateAlgoWorkspace(GenericAlgoWorkspace):
+    """Filesystem workspace for aggregate algo execution."""
```

Comment:
At the moment, the idea was to keep those to not introduce breaking changes within substra-tools knowing that this will be replaced by a single `@tools.function` decorator.

Reply:
I will remove the Generic terme so everything rely only on the `Algo` class though